Bug: Enum values lose their type information, causing return type mismatch errors

## Description

When returning an enum value from a function, the type checker reports a type mismatch error. Enum values lose their type information when accessed (e.g., `COLOR.RED`) and are treated as their underlying primitive type (`int`, `string`, `float`) instead of the enum type.

## Example EZ Code

### Example 1: Integer enum
```ez
import @std
using std

do main() {
  println(some_thing())
}

const COLOR enum {
  RED
  BLUE
  GREEN
}

do some_thing() -> COLOR {
  return COLOR.RED
}
```

**Error:**
```
error[E2005]: return type mismatch: expected COLOR, got int
  --> ./main.ez:6:21
   |
6 |   println(some_thing())
   |                     ^ wrong return type
   |

summary: generated 1 error
```

### Example 2: String enum (same issue)
```ez
const STATUS enum(string) {
  ACTIVE = "active"
  INACTIVE = "inactive"
}

do get_status() -> STATUS {
  return STATUS.ACTIVE  // Error: expected STATUS, got string
}
```

### Example 3: Float enum (same issue)
```ez
const PI_VARIANTS enum(float) {
  ROUGH = 3.14
  PRECISE = 3.14159
}

do get_pi() -> PI_VARIANTS {
  return PI_VARIANTS.ROUGH  // Error: expected PI_VARIANTS, got float
}
```

## Expected Behavior

The function should return successfully without type errors. When an enum value is accessed (e.g., `COLOR.RED`), it should:
1. Preserve its enum type information (`COLOR`)
2. Type checker should recognize `COLOR.RED` as type `COLOR`, not `int`
3. Return type validation should accept enum values as valid returns

```ez
do some_thing() -> COLOR {
  return COLOR.RED  // Should work - COLOR.RED is of type COLOR
}

do main() {
  temp c COLOR = some_thing()  // Should work
  println(c)  // Should print the enum value
}
```

## Actual Behavior

When an enum value is accessed:
1. The evaluator returns the underlying primitive value (Integer, String, or Float object)
2. Type information about the enum type is lost
3. Type checker sees the primitive type (`int`, `string`, `float`) instead of the enum type
4. Return type validation fails with E2005 type mismatch error

This happens for all enum base types: int (default), string, and float.

## Root Cause

Looking at `pkg/interpreter/evaluator.go:1422-1429`, when accessing an enum value:

```go
// Check for enum value access (e.g., STATUS.ACTIVE)
if enumObj, ok := obj.(*Enum); ok {
    if val, ok := enumObj.Values[node.Member.Value]; ok {
        return val  // Returns underlying primitive, loses enum type
    }
    // ...
}
```

The code returns the underlying value directly without preserving the enum type metadata.

## Proposed Solution

Create an `EnumValue` object type that wraps the underlying value while preserving the enum type:

### Changes needed in `pkg/interpreter/object.go`:

Add a new object type:
```go
type EnumValue struct {
    EnumType string  // e.g., "COLOR"
    Name     string  // e.g., "RED"
    Value    Object  // The underlying Integer/String/Float
}

func (ev *EnumValue) Type() ObjectType { return ENUM_VALUE_OBJ }
func (ev *EnumValue) Inspect() string {
    return fmt.Sprintf("%s.%s", ev.EnumType, ev.Name)
}
```

### Changes needed in `pkg/interpreter/evaluator.go`:

Update enum member access (around line 1425):
```go
if enumObj, ok := obj.(*Enum); ok {
    if val, ok := enumObj.Values[node.Member.Value]; ok {
        // Wrap the value to preserve enum type
        return &EnumValue{
            EnumType: enumObj.Name,
            Name:     node.Member.Value,
            Value:    val,
        }
    }
    // ...
}
```

### Changes needed in `pkg/interpreter/evaluator.go` (type checking):

Update `objectTypeToEZ()` around line 1224:
```go
func objectTypeToEZ(obj Object) string {
    switch v := obj.(type) {
    case *EnumValue:
        return v.EnumType  // Return "COLOR" not "int"
    case *Integer:
        return "int"
    // ...
    }
}
```

Update `typeMatches()` around line 1207 to handle EnumValue comparisons.

## Alternative Solutions

1. **Type annotations on primitives** - Add metadata to Integer/String/Float objects
   - Cons: Clutters primitive types with rarely-used fields
   
2. **Type checker inference** - Make type checker understand enum member access
   - Cons: Doesn't help with runtime type preservation, more complex

3. **Wrapper object** (recommended) - Dedicated EnumValue type
   - Pros: Clean separation, preserves type info, easy to unwrap for operations
   - Cons: Requires careful handling in operators (may need to unwrap for comparisons)

## Notes

- Enum values should preserve their type throughout the program lifecycle
- When used in operations (comparisons, etc.), should unwrap to underlying value
- Assignment to enum-typed variables should validate the enum type matches
- This affects variable declarations, function parameters, and return types
- Similar issue may exist with sized integer types (u8, i32) - see issue #75

--- a/pkg/interpreter/evaluator.go
+++ b/pkg/interpreter/evaluator.go
@@ -1240,6 +1240,9 @@ func objectTypeToEZ(obj Object) string {
 			return v.TypeName
 		}
 		return "struct"
+	case *EnumValue:
+		// Return the enum type name (e.g., "COLOR")
+		return v.EnumType
 	case *Nil:
 		return "nil"
 	case *Function:
@@ -1423,7 +1426,12 @@ func evalMemberExpression(node *ast.MemberExpression, env *Environment) Object {
 	// Check for enum value access (e.g., STATUS.ACTIVE)
 	if enumObj, ok := obj.(*Enum); ok {
 		if val, ok := enumObj.Values[node.Member.Value]; ok {
-			return val
+			// Wrap the value in an EnumValue to preserve type information
+			return &EnumValue{
+				EnumType: enumObj.Name,
+				Name:     node.Member.Value,
+				Value:    val,
+			}
 		}
 		return newErrorWithLocation("E3003", node.Token.Line, node.Token.Column,
 			"enum value '%s' not found in enum '%s'", node.Member.Value, enumObj.Name)

--- a/pkg/interpreter/object.go
+++ b/pkg/interpreter/object.go
@@ -28,6 +28,7 @@ const (
 	BREAK_OBJ        ObjectType = "BREAK"
 	CONTINUE_OBJ     ObjectType = "CONTINUE"
 	ENUM_OBJ         ObjectType = "ENUM"
+	ENUM_VALUE_OBJ   ObjectType = "ENUM_VALUE"
 )
 
 type Object interface {
@@ -210,6 +211,18 @@ func (e *Enum) Inspect() string {
 	}
 	out.WriteString(" }")
 	return out.String()
+}
+
+// EnumValue wraps an enum member value with its type information
+type EnumValue struct {
+	EnumType string // The enum type name (e.g., "COLOR")
+	Name     string // The value name (e.g., "RED")
+	Value    Object // The underlying value (Integer, Float, or String)
+}
+
+func (ev *EnumValue) Type() ObjectType { return ENUM_VALUE_OBJ }
+func (ev *EnumValue) Inspect() string {
+	return ev.Value.Inspect()
 }
 
 // Environment holds variable bindings

